### ISSUE TITLE
Support for Cose EC point compressed keys

### ISSUE THREAD

**Issue by n0900** [a-sit-plus/signum#22]
Currently not supported as both point de-/compression and serialization is missing

**Comment by JesusMcCloud** [comment]
TL;DR: Impossible, unless we make ugly changes to the CBOR serialization module

This is currently impossible due to the way the CBOR-Decoder works: When approaching the y-coordinate in the underlying CBOR-bytes, we need to make a choice, one of:

* try and decode a boolean (which is either `0xf4` or `0xf5`)
* or try and decode the beginning of an array (`0xf9`)

if we try to decode a bool and encounter `0xf9`, we have consumed the byte already and it is "lost", if we try to decode an array, and encounter a bool, we have also consumed a byte, which is lost. In both cases, an exception is thrown.

So we actually need to expose the `CborReader` (i.e. remove the `internal` modifier) to provide the ability for extension from outside the serialization module. We also need to expose `ByteArrayInput`, so we can reset the index after a failed decoding attempt.

Dirty business…

**Comment by JesusMcCloud** [comment]
wrt serialization:

- Encoding is easy: just encode  a byte array or a boolean
- Decoding is not pretty but also straight-forward in the end: since we use `CoseKey.Companion.deserialize` do kick off decoding we can just try twice: first we try to decode a point-compressed key and if that fails because a byte array was encountered, although a boolean was expected for the y-coordinate, we try again without point compression.
